File: server/pkg/adding/project.go
package adding

import "fmt"

// Project defines the adding form of a project entity.
type Project struct {
	Key               string `json:"key"`
	Name              string `json:"name"`
	Type              string `json:"type"`
	Description       string `json:"description,omitempty"`
	LeadID            string `json:"leadId,omitempty"`
	DefaultAssigneeID string `json:"defaultAssigneeId,omitempty"`
	DefaultBoardType  string `json:"defaultBoardType"`
}

func validateAddProject(p *Project) error {
	if p == nil {
		return fmt.Errorf("Project is nil")
	}
	if len(p.Key) == 0 {
		return fmt.Errorf("'key' is empty")
	}
	if len(p.Name) == 0 {
		return fmt.Errorf("'name' is empty")
	}
	if len(p.Type) == 0 {
		return fmt.Errorf("'type' is empty")
	}
	if len(p.DefaultBoardType) == 0 {
		return fmt.Errorf("'defaultBoardType' is empty")
	}

	return nil
}

File: client/src/transforms/project.js
export const addProjectTransform = data => {
  const {
    key,
    name,
    type,
    description,
    leadId,
    defaultAssigneeId,
    defaultBoardType,
  } = data;
  return {
    key,
    name,
    type,
    description,
    leadId,
    defaultAssigneeId,
    defaultBoardType,
  };
};

export const editSprintTransform = data => {
  return {
    ...data,
  };
};

export const startSprintTransform = data => {
  const { duration, ...sprint } = data;
  return {
    ...sprint,
  };
};
